File: iniciante/javascript/1118.js
var input = require('fs').readFileSync('stdin', 'utf8')
var lines = input.split('\n')


let media = 0
let countNotas = 0

while(true){

    let nota = parseFloat(lines.shift().replace('\r', ''))
    if(nota >= 0 && nota <= 10){
        media += nota
        countNotas++
    }else{
        console.log('nota invalida')
    }
    if(countNotas == 2){
        console.log(`media = ${(media/countNotas).toFixed(2)}`)
        do{
            console.log('novo calculo (1-sim 2-nao)')
            var menu = parseInt(lines.shift().replace('\r', ''))
        }while(menu != 1 && menu != 2 )
        
        if(menu == 1){
            countNotas = 0
            media = 0
        }else{
            break
        }
    }

}

File: iniciante/python/1118.py
# -*- coding: utf-8 -*-

count_notas_validas = 0
soma_notas = 0
novo_calculo = 0
flag_novo_calculo = False

while True:
    entrada = input()
    if '.' in entrada:
        entrada = float(entrada)
    else:
        entrada = int(entrada)

    if entrada >= 0 and entrada <= 10:
        soma_notas += entrada
        count_notas_validas += 1
    else:
        print('nota invalida')
    
    if count_notas_validas == 2:
        print(f'media = {(soma_notas / count_notas_validas):.2f}')
    
        while True:
            flag_novo_calculo = True
            print('novo calculo (1-sim 2-nao)')
            novo_calculo = int(input())
            if novo_calculo == 1 or novo_calculo == 2:
                break
    if novo_calculo == 1 and flag_novo_calculo:
        flag_novo_calculo = False
        soma_notas = 0
        count_notas_validas = 0
        novo_calculo = 0
    elif novo_calculo == 2:
        break
